### DISCUSSION TITLE
app Directory Cloudflare cache brake next.js application

### DISCUSSION THREAD

**Discussion by mikayel94** [vercel/next.js#45085] [App Router]
For example, 
https://example.com/blog URL has to different response

- response is an HTML with `content-type: text/html; charset=utf-8`
- response is a new react communication format like `J0:[[["",{"children":["blogs",{"children":["",{}]}]},null,null,true],"@1","@2"]] ...` with `content-type: application/octet-stream` 

so Cloudflare cache key is the same for both requests and as a result, when I reload the page it may be downloaded if the last cache in Cloudflare is the second response and vice versa.

to change the cache key template in Cloudflare need to be an Enterprise account maybe base on 
specific headers of the second request 
`next-router-prefetch: 1`
`next-router-state-tree: ["",{"children":["contact",{"children":["",{}]}]},null,null,true]`

next.js version is "next": "13.1.3" with app Directory

**Comment by leerob** [comment]
https://github.com/vercel/next.js/issues/48569
